sdk/storage/Azure.Storage.Blobs/src/BlockBlobWriteStream.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Shared;
+
+namespace Azure.Storage.Blobs
+{
+    internal class BlockBlobWriteStream : WriteStream
+    {
+        private readonly BlockBlobClient _blockBlobClient;
+        private readonly BlobRequestConditions _conditions;
+        private readonly List<string> _blockIds;
```

Comment:
Fixed in next iteration.

Reply:
I don't feel super comfortable with that behavior. If we aren't overwriting and can't get a block list to append to, I'd rather throw instead of having the one case where we blow your data away even when specified not to.

Reply:
Removing `BlockBlobOpenWriteOptions.Overwrite` in the next iteration, I intended to already.

Reply:
Do other streams default to false with that option? I'd expect them to, and I feel like that's a gotcha if this one case always overwrites.

Under that assumption: unless we make all the other write stream options `bool? Overwrite` and throw if they left it null, I think we need to make some sort of attempt to do an append if we want to provide a consistent user experience. The only downside I can think of is the extra service call we have to wait on before uploading.

I prefer the attempt to append, throwing when the operation is impossible. I dislike mandating users to set a `bool?` and it gives users more possibilities that would otherwise be denied.

Reply:
The other streams append by default.

@kasobol-msft, what do you think on this issue?  In track 1, `CloudBlockBlob.OpenWrite()` always overwrote the underlying blob.

Reply:
I'm looking at https://docs.microsoft.com/en-us/dotnet/api/system.io.file.openwrite?view=netcore-3.1 .
It seems that OpenWrite always overwrites if file exists. If user wants to append they're expected to use "File.AppendText". Therefore I think Overwrite should be default for all "OpenWrite" implementations.

Can we have a `bool` with reasonable default for `Overwrite`. I'm not sure we need 3-state boolean. https://thedailywtf.com/articles/tri-state-boolean

Reply:
After syncing offline, we decided to remove the overwrite parameter entirely.  All `WriteStreams` except `BlockBlobWriteStream` will append, and `BlockBlobWriteStream` will overwrite.